--- a/messages.rs
+++ b/messages.rs
@@ -1418,7 +1418,7 @@
         #[rasn(tag(context, 1), identifier = "listOfModifier")]
         pub list_of_modifier: Option<SequenceOf<Modifier>>,
         #[rasn(tag(context, 2), identifier = "confirmedServiceRequest")]
-        pub confirmed_service_request: ConfirmedServiceRequest,
+        pub confirmed_service_request: Box<ConfirmedServiceRequest>,
     }
     impl DefineEventActionRequest {
         pub fn new(
@@ -1429,7 +1429,7 @@
             Self {
                 event_action_name,
                 list_of_modifier,
-                confirmed_service_request,
+                confirmed_service_request: Box::new(confirmed_service_request),
             }
         }
     }
@@ -2546,7 +2546,7 @@
         #[rasn(tag(context, 1), identifier = "listOfModifier")]
         pub list_of_modifier: SequenceOf<Modifier>,
         #[rasn(tag(context, 2), identifier = "confirmedServiceRequest")]
-        pub confirmed_service_request: ConfirmedServiceRequest,
+        pub confirmed_service_request: Box<ConfirmedServiceRequest>,
     }
     impl GetEventActionAttributesResponse {
         pub fn new(
@@ -2557,7 +2557,7 @@
             Self {
                 mms_deletable,
                 list_of_modifier,
-                confirmed_service_request,
+                confirmed_service_request: Box::new(confirmed_service_request),
             }
         }
     }
@@ -5334,14 +5334,14 @@
         #[rasn(tag(context, 1), identifier = "numberOfElements")]
         pub number_of_elements: Unsigned32,
         #[rasn(tag(context, 2), identifier = "elementType")]
-        pub element_type: TypeSpecification,
+        pub element_type: Box<TypeSpecification>,
     }
     impl TypeDescriptionArray {
         pub fn new(packed: bool, number_of_elements: Unsigned32, element_type: TypeSpecification) -> Self {
             Self {
                 packed,
                 number_of_elements,
-                element_type,
+                element_type: Box::new(element_type),
             }
         }
     }
